app/controllers/users/sessions_controller.rb

```diff
@@ -9,21 +11,35 @@ def new
   # POST /resource/sign_in
   def create
     super
-    if user_signed_in?
-      UserAction.save_user_action(current_user.id,  UserAction::ACTION_SIGN_IN, new_user_session_path)
-    end
   end
 
   # DELETE /resource/sign_out
   def destroy
-    UserAction.save_user_action(current_user.id,  UserAction::ACTION_SIGN_OUT, UserAction::PATH_SIGN_OUT)
     super
   end
 
-  # protected
+  private
+    def user_log_in
+      if current_user.present?
+        UserAction.save_user_action(current_user.id,  UserAction::ACTION_SIGN_IN, new_user_session_path)
+        cookies[:login_attempts] = 0
+      end
+    end
 
-  # If you have extra params to permit, append them to the sanitizer.
-  # def configure_sign_in_params
-  #   devise_parameter_sanitizer.permit(:sign_in, keys: [:attribute])
-  # end
+    def user_sign_out
+      UserAction.save_user_action(current_user.id,  UserAction::ACTION_SIGN_OUT, UserAction::PATH_SIGN_OUT)
+    end
+
+    def recaptcha_check
+      cookies[:login_attempts] ||= 0
+      cookies[:login_attempts] = cookies[:login_attempts].to_i + 1
+      if cookies[:login_attempts].to_i > 3
```

Comment:
avoid of using `if` without `else` statement
look above

Reply:
Fixed